libs/gen/EnumsSourceGenerator.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Garnet;
+
+[Generator]
+public class EnumsSourceGenerator : IIncrementalGenerator
+{
+    const string GeneratedClassName = "EnumUtils";
+
+    public void Initialize(IncrementalGeneratorInitializationContext context)
+    {
+        var enumDetails = context.SyntaxProvider
+            .CreateSyntaxProvider(
+                predicate: static (node, _) => node is EnumDeclarationSyntax enumDeclarationSyntax
+                    && enumDeclarationSyntax.AttributeLists.Any(al => al.Attributes.Any(a => a.Name.ToString() == "GenerateEnumDescriptionUtils")),
+                transform: static (ctx, _) => TransformEnumDetails((EnumDeclarationSyntax)ctx.Node, ctx.SemanticModel)
+            );
+
+        var enumUtils = enumDetails.Select((details, _) => Execute(details));
+
+        context.RegisterSourceOutput(enumUtils, (ctx, source) => ctx.AddSource($"{GeneratedClassName}.{source.EnumName}.g.cs", source.ClassSource));
+    }
+
+    private static EnumDetails TransformEnumDetails(EnumDeclarationSyntax enumDeclaration, SemanticModel semanticModel)
+    {
+        var namespaceDeclaration = enumDeclaration.FirstAncestorOrSelf<NamespaceDeclarationSyntax>();
+        var enumName = enumDeclaration.Identifier.Text;
+
+        var values = enumDeclaration.Members
+            .Select(m => (
+                m.Identifier.Text,
+                semanticModel.GetOperation(m.EqualsValue!.Value)!.ConstantValue.Value,
+                Description: m.AttributeLists
+                    .SelectMany(al => al.Attributes).Where(a => a.Name.ToString() == "Description")
+                    .SingleOrDefault()
+            ))
+            .ToList();
+        return new EnumDetails(namespaceDeclaration!.Name.ToString(), enumName, values);
+    }
+
+    private static (string EnumName, string ClassSource) Execute(EnumDetails details)
+    {
+        var classBuilder = new StringBuilder();
```

Comment:
I highly recommend using [IndentedTextWriter](https://learn.microsoft.com/en-us/dotnet/api/system.codedom.compiler.indentedtextwriter?view=net-8.0) for syntax writing and get rid of the manual indenting which tends to get unmaintainable really fast.